### PR TITLE
Fix and improve the export function for Bioimage.IO

### PR DESCRIPTION
**Pull request by qin-yu**
The current `export_bioimageio()` function

- generates an _environment.yaml_ file containing an erroneous line of code, leading to installation errors. 
  - #252 
  - https://github.com/bioimage-io/collection-bioimage-io/pull/649
- generates an unnecessary  _environment.yaml_ file and add to _rdf.yaml_
  - #253 
    a default config file led new users to take it for granted, resulting in unwanted reuploads and DOI records on Zenodo, and unnecessary installations of environments for bioimage.io's CI on GitHub.
  - https://github.com/bioimage-io/collection-bioimage-io/pull/649

### CODE REVIEW

**Comment by uschmidt83** [comment]
I don't remember why we generate a conda environment file, maybe @constantinpape or @esgomezm remember. See #171 for the history of how the bioimage.io export came to be.

Sorry for the very late reply.

**Comment by esgomezm** [comment]
Hi @uschmidt83 , I don't know why the environment was included here. It was @constantinpape who took care in this case. Let me discuss with the Ilastik team if it was a way of integrating StarDist or something, and get back to you

**Comment by FynnBe** [comment]
I'm speculating, but maybe it was intended to document a conda env in which the postprocessing could also be executed. 
As bioimageio does not support this kind of arbitrary postprocessing and exported models are pure tensorflow afaik, stardist models don't need a conda env if the `tensorflow_version` is set.

**Comment by qin-yu** [comment]
Hey @uschmidt83, it appears that Constantin may not be available to provide further explanations. Based on our discussions, we seem to agree that the config is unnecessary and my changes make sense. Could you please review again? Let me know if there are any comments, or if we're ready to merge or close the PR. Many thanks~

**Comment by constantinpape** [comment]
From what I recall the conda env was indeed there to install the required dependencies for post-processing. So it's not really necessary for just running the model.

**Comment by uschmidt83** [comment]
Ok, so the consensus seems to be that the `environment.yaml` file is unnecessary for StarDist (since it's a TensorFlow-based model), but it still somehow causes trouble for the Bioimage.io CICD pipeline. Is this because other types of model exports must create `environment.yaml`?

File: stardist/bioimageio_utils.py
Comment (uschmidt83) [inline stardist/bioimageio_utils.py:79]:
```suggestion
    reqs_conda = [f"{req.project_name}{req.specifier}" for req in pkg_info.requires(extras=['bioimageio']) if req.key.startswith('bioimageio')]
```

File: stardist/bioimageio_utils.py
Comment (uschmidt83) [inline stardist/bioimageio_utils.py:349]:
```suggestion
    overwrite_spec_kwargs=None,
    generate_default_deps=False,
```

File: stardist/bioimageio_utils.py
Comment (uschmidt83) [inline stardist/bioimageio_utils.py:381]:
```suggestion
    overwrite_spec_kwargs: dict or None
        spec keywords that should be overloaded (default: None)
    generate_default_deps: bool
        not required for bioimage.io, i.e. StarDist models don't need a dependencies field in rdf.yaml (default: False)
        if True, generate an environment.yaml file recording the python, bioimageio.core, stardist and tensorflow requirements
        from which a conda environment can be recreated to run this export
```

**Review by uschmidt83 — CHANGES_REQUESTED** [review]
Can you please just make these small changes.

**Comment by uschmidt83** [comment]
Apparently, I'm allowed to commit these changes myself... GitHub's PR system is confusing.

**Comment by qin-yu** [comment]
Hey @uschmidt83 I believe environment.yaml is not required at all unless you want to provide one for your users. But I've been waiting for @FynnBe's confirmation before commenting here. But I guess you finalised the PR already.

> Apparently, I'm allowed to commit these changes myself... GitHub's PR system is confusing.

Yes it could be confusing, I never thought about this. I am reading [About permissions of forks](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/about-permissions-and-visibility-of-forks#about-permissions-of-forks) to understand it better.

File: stardist/bioimageio_utils.py
Comment (qin-yu) [reply stardist/bioimageio_utils.py]: Oops, should have kept the order, thanks

**Comment by uschmidt83** [comment]
I will merge this PR unless someone (@FynnBe, ?) objects.

**Pull request merged**
